api/imap-client/imap_client.py

```
@@ -179,14 +211,22 @@ def _get_attachments(self, message_data: Message):
             logging.info(f"Filename: {filename} mime_type: {mime_type}")
 
-            if len(filename) > 0 and self.valid_mime_type(mime_type):
-                filePath = os.path.join(base_path, "temp", filename)
+            if filename and len(filename) > 0 and self.valid_mime_type(mime_type):
+                # Sanitize: the filename comes from untrusted email content and
+                # is used in a path join, so strip any directory components to
+                # prevent path traversal (e.g. "../../etc/...").
+                safe_filename = os.path.basename(filename)
+                payload = part.get_payload(decode=True)
+                if payload is None:
+                    logging.warning(f"Skipping attachment with undecodable payload: {safe_filename}")
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)